### PR TITLE
IAM related changes

### PR DESCRIPTION
**Pull request by sanjeevghimire**
* code change to accommodate both username+password and IAM based authentication
* readme changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,18 +146,33 @@ wsk action create iot-pub iot-pub.py
...
 Once the actions are successfully created, we can set default service credentials for each of the actions. Otherwise we’d have to pass in the service credentials every time we’d like our actions to call the Watson services. To obtain these credentials, click each provisioned service in the IBM Cloud dashboard, and then select the `View credentials` dropdown.
 
-<p align="center">
-<img src="/images/stt_creds.png" data-canonical-src="/images/stt_creds.png" width="600" height="400" style="margin-left: auto; margin-right: auto;" />
-</p>
+![](images/stt_creds.png)
 
 Then insert the corresponding credentials when running the commands below.
 
+* If the service credentials from IBM Watson Assistant is username/password based as below populate the username, password and workspace_id and comment out the IAM credentials part.
```
Comment (tanmayb123) [inline README.md]: Maybe change this to something along the lines of:
"If the service credentials from IBM Watson Assistant are username/password as shown below, populate the username, password, and workspace_id fields, and comment out the IAM credentials field."

File: README.md [hunk 2]
```diff
@@ -150,18 +146,33 @@ wsk action create iot-pub iot-pub.py
...
+![](https://github.com/IBM/pattern-images/raw/master/watson-assistant/WatsonAssistantCredentials.png)
+
+
 ```
 wsk action update conversation -p username ${conversation_username} -p password ${conversation_password} -p workspace_id ${conversation_workspace_id}
 wsk action update iot-pub -p iot_org_id ${iot_org_id} -p device_id ${device_id} -p api_token ${api_token} -p device_type ${device_type}
 wsk package bind /whisk.system/watson-speechToText myWatsonSpeechToText -p username ${stt_username} -p password ${stt_password}
 ```
 
+* If the service credentials from IBM Watson Assistant is IAM based as below, populate the IAM apikey, url, and workspace_id and comment out the username/password part
```
Comment (tanmayb123) [inline README.md]: Suggestion:
"If the service credentials from IBM Watson Assistant are IAM based as below, populate the IAM apikey, url, and workspace_id, and comment out the username/password fields."

**Review by tanmayb123 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -56,14 +56,10 @@ We can get started by assembling and configuring the RF circuit. This circuit re
 
 Once all components have been obtained, assemble them to form the circuit below. In this circuit, we have the Raspberry Pi connected to the electronic breadboard via the GPIO ribbon/breakout board.
 
-![Circuit](/images/home_automation_labeled.png "CircuitFritzing")
+![Circuit](images/home_automation_labeled.png "CircuitFritzing")
 
 <!-- Transmitter/Receiver datasheet http://www.mantech.co.za/Datasheets/Products/433Mhz_RF-TX&RX.pdf -->
```
Comment (scottdangelo) [inline README.md:62]: Did you mean to delete this image?

**Review by scottdangelo — COMMENTED** [review]
Wondering about a deleted link to an image...

**Comment by scottdangelo** [comment]
OK, was just a comment LGTM

File: ibm_cloud_functions/conversation.js [hunk 4]
```diff
@@ -17,27 +17,33 @@ var request = require('request');
...
+   var auth = {"user": username,"pass":password}
+   var url = "https://gateway.watsonplatform.net/assistant/api/v1/workspaces/" + workspace_id + "/message?version=2018-07-10"
+   if(iamApiKey){
+    auth = {"user": "apikey","pass":iamApiKey}
+    url = "https://gateway-wdc.watsonplatform.net/assistant/api/v1/workspaces/" + workspace_id + "/message?version=2018-07-10"
+   }   
    var input_text = params.data
-   var url = "https://gateway.watsonplatform.net/conversation/api/v1/workspaces/" + workspace_id + "/message?version=2017-04-21"
    var body = {"input": {"text": input_text}}
+   console.log(auth)
```
Comment (kkbankol-ibm) [inline ibm_cloud_functions/conversation.js]: can you remove this print statement?

Comment (sanjeevghimire) [reply ibm_cloud_functions/conversation.js]: done @kkbankol-ibm

**Review by kkbankol-ibm — APPROVED** [review]

**Pull request merged**